flink-runtime/src/main/java/org/apache/flink/runtime/externalresource/ExternalResourceUtils.java

```diff
@@ -0,0 +1,213 @@
+
+		final Map<String, ExternalResourceDriver> externalResourceDrivers = new HashMap<>();
+		for (String resourceName: resourceSet) {
+			final ConfigOption<String> driverClassOption =
+				key(ExternalResourceOptions.keyWithResourceNameAndSuffix(resourceName, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_FACTORY_SUFFIX))
+					.stringType()
+					.noDefaultValue();
+			final String driverFactoryClassName = config.getString(driverClassOption);
+			if (StringUtils.isNullOrWhitespaceOnly(driverFactoryClassName)) {
+				LOG.warn("Could not find driver class name for {}. Please make sure {} is configured.",
+					resourceName, driverClassOption.key());
+				continue;
+			}
+
+			ExternalResourceDriverFactory externalResourceDriverFactory = externalResourceFactories.get(driverFactoryClassName);
+			if (externalResourceDriverFactory != null) {
+				DelegatingConfiguration delegatingConfiguration =
+					new DelegatingConfiguration(config, ExternalResourceOptions.keyWithResourceNameAndSuffix(resourceName, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_PARAM_SUFFIX));
+				try {
+					externalResourceDrivers.put(resourceName, externalResourceDriverFactory.createExternalResourceDriver(delegatingConfiguration));
+					LOG.info("Add external resources driver for {}.", resourceName);
+				} catch (Exception e) {
+					LOG.warn("Could not instantiate driver with factory {} for {}. {}", driverFactoryClassName, resourceName, e);
+				}
+			} else {
+				LOG.warn("Could not find factory class {} for {}.", driverFactoryClassName, resourceName);
+			}
+		}
+
+		return externalResourceDrivers;
+	}
+
+	/**
+	 * Instantiate the {@link ExternalResourceInfoProvider} for all of enabled external resources.
+	 */
+	public static ExternalResourceInfoProvider getExternalResourceInfoProvider(Configuration config, PluginManager pluginManager) {
+		final Map<String, ExternalResourceDriver> externalResourceDrivers = externalResourceDriversFromConfig(config, pluginManager);
+		final Map<String, Long> externalResourceAmountMap = getExternalResourceAmountMap(config);
+		final Map<String, Set<? extends ExternalResourceInfo>> externalResources = new HashMap<>();
+		for (Map.Entry<String, ExternalResourceDriver> externalResourceDriverEntry : externalResourceDrivers.entrySet()) {
+			final String resourceName = externalResourceDriverEntry.getKey();
+			final ExternalResourceDriver externalResourceDriver = externalResourceDriverEntry.getValue();
+			if (externalResourceAmountMap.containsKey(resourceName)) {
+				try {
+					final Set<? extends ExternalResourceInfo> externalResourceInfos;
+					externalResourceInfos = externalResourceDriver.retrieveResourceInfo(externalResourceAmountMap.get(resourceName));
+					externalResources.put(resourceName, externalResourceInfos);
+				} catch (Exception e) {
+					LOG.error("Failed to retrieve information of external resource {}. {}", resourceName, e);
```

Comment:
```suggestion
					LOG.warn("Failed to retrieve information of external resource {}.", resourceName, e);
```

you don't have to specify a place holder for exceptions. They will be still logged.